avaframe/com1DFA/deriveParameterSet.py

```diff
@@ -354,45 +368,58 @@ def setThicknessValueFromVariation(key, cfg, simType, row):
-            for count, id in enumerate(idList):
-                thNameId = thType + id
-                cfg['GENERAL'][thNameId] = str(float(thicknessList[count]) * variationFactor)
-
-            # set percentVaration parameter to actual variation in percent$steps
-            if variationFactor == 1.:
-                # if variation is 0% set percentVaration = ''
-                variationIni = ''
-            elif variationFactor < 1:
-                variationIni = '-' + str((1. - variationFactor) * 100) + '$1'
-            elif  variationFactor > 1:
-                variationIni = '+' + str((variationFactor - 1.) * 100) + '$1'
-
-            # update parameter value
-            cfg['GENERAL'][key] = variationIni
+    for varType in ['Range', 'Percent']:
+        entCondition = (key == ('entTh%sVariation' % varType) and 'ent' in simType)
+        secRelCondition = (key == ('secondaryRelTh%sVariation' % varType) and cfg['GENERAL']['secRelArea'] == 'True')
+        relCondition = (key == ('relTh%sVariation' % varType))
+
+        # fetch variation factor
+        variationFactor = float(row._asdict()[key])
+
+        # update thickness values according to variation
+        if entCondition or secRelCondition or relCondition:
+            thType = key.split(varType)[0]
+            thFlag = thType + 'FromShp'
+
+            # add thickness values for all features if thFromShape = True
+            if cfg['GENERAL'][thFlag] == 'True':
+                thId = thType + 'Id'
+                thThickness = thType + 'Thickness'
+                thicknessList = cfg['INPUT'][thThickness].split('|')
+                idList = cfg['INPUT'][thId].split('|')
+                for count, id in enumerate(idList):
+                    thNameId = thType + id
+
+                    if varType == 'Percent':
+                        cfg['GENERAL'][thNameId] = str(float(thicknessList[count]) * variationFactor)
+                        # set percentVaration parameter to actual variation in percent$steps
+                        if variationFactor == 1.:
+                            # if variation is 0% set percentVaration = ''
+                            variationIni = ''
+                        elif variationFactor < 1:
+                            variationIni = '-' + str((1. - variationFactor) * 100) + '$1'
+                        elif  variationFactor > 1:
+                            variationIni = '+' + str((variationFactor - 1.) * 100) + '$1'
+                    elif varType == 'Range':
+                        cfg['GENERAL'][thNameId] = str(float(thicknessList[count]) + variationFactor)
+                        # set percentVaration parameter to actual variation in percent$steps
+                        if variationFactor == 0.:
+                            # if variation is 0 set RangeVaration = ''
+                            variationIni = ''
+                        else:
+                            variationIni = str(variationFactor) + '$1'
```

Comment:
I think this is now one step too much on the nesting and should be rewritten with subfunctions. Maybe seperate the vartypes?

Reply:
i made subfunction for the creation of varIni